src/overlays/actors/ovl_Bg_Treemouth/z_bg_treemouth.c

```diff
@@ -21,29 +37,215 @@ const ActorInit Bg_Treemouth_InitVars = {
+    }
+}
+
+void func_808BC864(BgTreemouth* this, GlobalContext* globalCtx) {
+    this->unk_168 -= 0.03f;
+    if (this->unk_168 <= 0.0f) {
+        BgTreemouth_SetupAction(this, func_808BC65C);
+    }
+}
+
+void func_808BC8B8(BgTreemouth* this, GlobalContext* globalCtx) {
+    if ((!(Flags_GetEventChkInf(5))) || (LINK_IS_ADULT)) {
+        if (LINK_IS_CHILD) {
+            if (Flags_GetEventChkInf(0xC)) {
+                if (func_8002E12C(&this->dyna.actor, 1658.0f, 0x7530)) {
+                    this->dyna.actor.flags |= 1;
+                    if (this->dyna.actor.unk_10C != 0) {
+                        this->dyna.actor.flags &= ~1;
+                        globalCtx->csCtx.segment = D_808BD2A0;
+                        gSaveContext.cutsceneTrigger = 1;
+                        BgTreemouth_SetupAction(this, func_808BC9EC);
+                    }
+                }
+            } else if (func_8002E12C(&this->dyna.actor, 1658.0f, 0x4E20)) {
+                Flags_SetEventChkInf(0xC);
+                globalCtx->csCtx.segment = D_808BCE20;
+                gSaveContext.cutsceneTrigger = 1;
+                BgTreemouth_SetupAction(this, func_808BC9EC);
+            }
+        }
+    } else {
+        this->unk_168 = 1.0f;
+    }
+}
+
+void func_808BC9EC(BgTreemouth* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (globalCtx->csCtx.state == 3) {
+        if (func_8002E12C(&this->dyna.actor, 350.0f, 0x7530)) {
+            player->actor.posRot.pos.x = 3827.0f;
+            player->actor.posRot.pos.y = -161.0f;
+            player->actor.posRot.pos.z = -1142.0f;
+        }
+
+        globalCtx->csCtx.frames = 0;
+        globalCtx->csCtx.unk_18 = 0xFFFF;
+        D_8015FCC0 = 0xFFFF;
+        D_8015FCC2 = 0xFFFF;
+        D_8015FCC4 = 0xFFFF;
+        globalCtx->csCtx.unk_1A = 0;
+        globalCtx->csCtx.unk_1B = 0;
+        globalCtx->csCtx.state = 2;
+
+        if (globalCtx->msgCtx.choiceIndex == 0) {
+            globalCtx->csCtx.segment = D_808BD520;
+            Flags_SetEventChkInf(5);
+            BgTreemouth_SetupAction(this, func_808BCAF0);
+        } else {
+            globalCtx->csCtx.segment = D_808BD790;
+            globalCtx->csCtx.frames = 0;
+            BgTreemouth_SetupAction(this, func_808BC8B8);
+        }
+    }
+}
+
+void func_808BCAF0(BgTreemouth* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* actorAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        actorAction = globalCtx->csCtx.npcActions[0];
+        if (actorAction != NULL) {
+            if (actorAction->action == 2) {
+                BgTreemouth_SetupAction(this, func_808BC80C);
+            } else if (actorAction->action == 3) {
+                Audio_PlaySoundGeneral(NA_SE_EV_WOODDOOR_OPEN, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                BgTreemouth_SetupAction(this, func_808BC6F8);
+            }
+        }
+    }
+}
+
+void func_808BCB8C(BgTreemouth* this, GlobalContext* globalCtx) {
+}
```

Comment:
i name these functions _DoNothing, as long as it makes sense for that actor. given this gets set after the deku tree mouth is fully open i think its a good idea to name it this